src/main/java/edu/ucsb/cs156/organic/controllers/StaffController.java

```diff
@@ -0,0 +1,116 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.ucsb.cs156.organic.entities.Staff;
+import edu.ucsb.cs156.organic.entities.User;
+import edu.ucsb.cs156.organic.errors.EntityNotFoundException;
+import edu.ucsb.cs156.organic.repositories.StaffRepository;
+import edu.ucsb.cs156.organic.repositories.UserRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+
+@Tag(name = "Staff")
+@RequestMapping("/api/staff")
+@RestController
+@Slf4j
+public class StaffController extends ApiController{
+
+    @Autowired
+    StaffRepository staffRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Operation(summary = "List all staff")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @GetMapping("/all")
+    public Iterable<Staff> allCourses() {
+        User u = getCurrentUser().getUser();
+        log.info("u={}", u);
+
+        return staffRepository.findAll();
+    }
+
+    @Operation(summary = "Create a new staff")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @PostMapping("")
+    public Staff postCourse(
+            @Parameter(name = "courseId", description = "course id") @RequestParam Long courseId,
+            @Parameter(name = "githubId", description = "school abbreviation e.g. UCSB") @RequestParam Integer githubId)
+            throws JsonProcessingException {
+
+
+        Staff staff = Staff.builder()
+                    .courseId(courseId)
+                    .githubId(githubId)
+                    .build();
+
+        Staff savedStaff = staffRepository.save(staff);
+
+        return savedStaff;
+    }
+
+    @Operation(summary = "Get Staff")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @GetMapping("")
+    public Staff getStaff(
+            @Parameter(name = "id") @RequestParam Long id)
+            throws JsonProcessingException {
+
+        Staff staff = staffRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Staff.class, id.toString()));
+
+        return staff;
+    }
+
+    @Operation(summary = "Update information for a staff member")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @PutMapping("")
+    public Staff updateStaff(
+            @Parameter(name = "id") @RequestParam Long id,
+            @Parameter(name = "courseId", description = "course id") @RequestParam Long courseId,
+            @Parameter(name = "githubId", description = "school abbreviation e.g. UCSB") @RequestParam Integer githubId)
+            throws JsonProcessingException {
+
+        Staff staff = staffRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Staff.class, id.toString()));
+
+        staff.setCourseId(courseId);
```

Comment:
No... don't allow editing of these fields.  The only thing we should be doing here is allowing a user to delete a staff member.  If you are changing the github id or the course id, that should be a whole other record.  The reason is that staff ids might be associated with the audit trail of what someone did in the app. You would not want an admin to change the github id associated with a staff member from Alice to Bob and suddenly Bob is responsible for everythign Alice may have done.

This is another reason to put these operations in as separate PRs.

Reply:
Basically: just remove PUT / Edit completely.  We will only have POST, GET and DELETE for staff members (for now).

Reply:
Ok i removed the Edit and modified the test cases